Library/Homebrew/livecheck.rb

```diff
@@ -158,13 +164,24 @@ def throttle(rate = T.unsafe(nil))
   # `@url` instance variable when no argument is provided. The argument can be
   # a `String` (a URL) or a supported `Symbol` corresponding to a URL in the
   # formula/cask/resource (e.g. `:stable`, `:homepage`, `:head`, `:url`).
+  # Any options provided to the method are passed through to `Strategy` methods
+  # (`page_headers`, `page_content`).
   sig {
     params(
       # URL to check for version information.
-      url: T.any(String, Symbol),
+      url:     T.any(String, Symbol),
+      options: T.untyped,
     ).returns(T.nilable(T.any(String, Symbol)))
   }
-  def url(url = T.unsafe(nil))
+  def url(url = T.unsafe(nil), **options)
+    if options.present?
+      if @url_options.present?
+        @url_options.merge!(options)
```

Comment:
I've set this up to merge values with an existing `@url_options` hash but I'm not sure if this behavior would be surprising (or desirable) to users. We may want to simply overwrite the existing `@urls_options` if new `options` are provided and leave it up to the caller to merge `livecheck.url_options` with their hash if that's the behavior they want.

This is unlikely to be a concern for first-party `livecheck` blocks (as we only call `url` with arguments once in formulae/casks/resources), so overwriting may be a more simple and straightforward mental model anyway.